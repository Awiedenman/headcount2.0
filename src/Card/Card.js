import React, { Component } from 'react';
import PropTypes from 'prop-types'
import './Card.css'

class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      clicked: true
    }

  }

  handleCardClick = () => {
    this.setState({ clicked: !this.state.clicked });
    this.props.updateCompareState(this.state.clicked, this.props);
  }

  render() {

    let districtKeys = Object.keys(this.props.district.stats);
    const listItems = districtKeys.map(( stat, index ) => {
      return (
        <li 
        key={ index }
          className={ this.props.district.stats[stat] > 0.5 ? 'greater-than' : 'less-than'}
        > { `${stat}: ${this.props.district.stats[stat]}` } </li>
      )
    });
  
    return (
      <div className="card" onClick={ this.handleCardClick }> 
        <h3>
          { this.props.district.location }
        </h3>
        <ul>
          { listItems }
        </ul>
      </div>
    )
  }
}

Card.propTypes = {
  location: PropTypes.string,
  stats: PropTypes.arrayOf(PropTypes.object)
}



export default Card;