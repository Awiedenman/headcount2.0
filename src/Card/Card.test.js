import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Card from './Card.js';

describe('Card', () => {
  let card;
  const mockData = { 
    location: "Colorado", 
    stats: { "2007": 0.013}
  };
  const mockCards = [
    {district: {location: "COLORADO", stats: { "2007": 0.013}}}
  ];
 
  beforeEach(() => {
    card = shallow(< Card 
      district={ mockData }
      updateCompareState={ jest.fn() }
      comparedCards = {mockCards} />);
  });

    it('should match the snapshot', () => {
      expect(card).toMatchSnapshot();
    });

   it('should render a card with output from the  location props that it was passed', () => {
     const expectation = 'Colorado';

     expect(card.find('h3').text()).toBe(expectation);
    });

   it('should render a card with output from the stats props that it was passed', () => {
     const expectation = ' 2007: 0.013 ';

     expect(card.find('ul').childAt(0).text()).toBe(expectation);
    });

    it('should call handleCardClick when the card is clicked', () => {
      card = mount(< Card 
        district={ mockData }
        updateCompareState={ jest.fn() }
        comparedCards = { mockCards } />);
      const spy = spyOn(card.instance(), 'handleCardClick')
      card.instance().forceUpdate();

      card.find('div').simulate('click');

      expect(spy).toHaveBeenCalled();
      
    });

    it('should not show 2 of the same cards', () => {
      let mockCards = [
        {district: {location: "COLORADO", stats: { "2007": 0.013}}},
        {district: {location: "COLORADO", stats: { "2007": 0.013}}}
      ];

      expect(card.instance().handleCardClick()).toBe(undefined)


    });

    it('should update clicked state to opposite boolean when handleCardClick is called', () => {
      card.setState({clicked: false})

      card.instance().handleCardClick();

      expect(card.state('clicked')).toBe(true);
    });

    it.skip('should call updateCompareState method from App component when handleCardClick is called', () => {
      card = mount(< Card 
        district={ mockData }
        updateCompareState={ jest.fn() }
        comparedCards = { mockCards } />);
      const spy = spyOn(card.instance(), 'updateCompareState')
      
      card.instance().forceUpdate();

      card.find('div').simulate('click');

      expect(spy).toHaveBeenCalled();
    });
});


 
