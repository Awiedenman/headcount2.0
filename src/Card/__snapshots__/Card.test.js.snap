// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    comparedCards={
        Array [
            Object {
              "district": Object {
                "location": "COLORADO",
                "stats": Object {
                  "2007": 0.013,
                },
              },
            },
          ]
    }
    district={
        Object {
            "location": "Colorado",
            "stats": Object {
              "2007": 0.013,
            },
          }
    }
    updateCompareState={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Colorado
</h3>,
        <ul>
          <li
                    className="less-than"
          >
                     
                    2007: 0.013
                     
          </li>
</ul>,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Colorado",
        },
        "ref": null,
        "rendered": "Colorado",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="less-than"
>
               
              2007: 0.013
               
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2007: 0.013",
                " ",
              ],
              "className": "less-than",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2007: 0.013",
              " ",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Colorado
</h3>,
          <ul>
            <li
                        className="less-than"
            >
                         
                        2007: 0.013
                         
            </li>
</ul>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Colorado",
          },
          "ref": null,
          "rendered": "Colorado",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="less-than"
>
                 
                2007: 0.013
                 
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2007: 0.013",
                  " ",
                ],
                "className": "less-than",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2007: 0.013",
                " ",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
